src/server/task_manager/task_pool/task_manager.ts

```diff
@@ -35,16 +35,17 @@ export class TaskManager {
     this.store = opts.store;
   }
 
-  public async schedule(task: TaskInstance) {
-    await this.store.schedule(task);
-    this.poller.attemptWork();
+  public async schedule(task: TaskInstance): Promise<RawTaskDoc> {
+    const result = await this.store.schedule(task);
+    this.poller.attemptWork(); // TODO await this?
```

Comment:
I don't think it's necessary to await this, as the attemptWork won't fail (it swallows / logs errors). And there's no reason to further delay the call to `schedule`.